Modified the path again

Medha-B opened the pull request:
Pull request for modified paths.

Medha-B committed:
Modified the path again

kaitoii11 commented:
Much better! However, why do we want to use `System.getProperty("user.home")` over `   System.getProperty("user.dir")`.
I thought we discussed this [here](https://github.com/Medha-B/unetdl4j/issues/2#issuecomment-663820424).

Medha-B commented:
I am sorry I didn't read carefully. Will make the necessary changes. 
Regarding adding the datasets: Should I make a folder in this repository for datasets, add them and just edit the gitignore file by putting the folder name? If I do this and then rebase in Eclipse, will it work? Or will the entire process be done using the command prompt?

Medha-B committed:
Create toDelete.java

Medha-B committed:
Add files via upload

Medha-B committed:
Add files via upload

Medha-B committed:
Delete toDelete.java

Medha-B committed:
Update .gitignore

kaitoii11 commented:
> Regarding adding the datasets: Should I make a folder in this repository for datasets, add them and just edit the gitignore file by putting the folder name? 

Yes. 

> If I do this and then rebase in Eclipse, will it work? Or will the entire process be done using the command prompt?
I don't think you need to rebase. You just have to commit your change and you should be fine.


Also, I see you have added `.gitignore` but you added the images on commits prior to this one. Which means the image is still tracked in git. You should remove these images.
https://docs.github.com/en/github/managing-large-files/removing-files-from-a-repositorys-history

Medha-B commented:
So, .gitignore should be edited before adding the datasets? In that case, I would have to delete the datasets, let the gitignore remain as it is, and then again upload the datasets?
Is this right?

kaitoii11 commented:
> So, .gitignore should be edited before adding the datasets?

Yes. Or at the same time.

>  In that case, I would have to delete the datasets, let the .gitignore remain as it is, and then again upload the datasets?

You just have to delete the datasets index within git (meaning you don't literally have to delete the file on your local PC).
This [article](https://docs.github.com/en/github/managing-large-files/removing-files-from-a-repositorys-history#removing-a-file-added-in-the-most-recent-unpushed-commit) might be a help.  Assuming you're following this article, you don't have to use the `--amend -CHEAD` option instead, just add it as a new commit.
So your steps should be something like this.
```bash
$ git rm --cached -r dataset # -r option means that it will run recursively
$ git commit -m "delete dataset folder"   # you can edit the commit message as you like
```

Medha-B committed:
Delete dataset

Medha-B commented:
Thanks for helping! Does this look correct?
Should I make a new folder for dataset now (same as before so that .gitignored need not be changed)?

kaitoii11 commented:
Yes, Good job!
> Should I make a new folder for dataset now (same as before so that .gitignored need not be changed)?

What do you mean by a new folder?  Do you mean in your local folder?

> Much better! However, why do we want to use `System.getProperty("user.home")` over ` System.getProperty("user.dir")`.
> I thought we discussed this [here](https://github.com/Medha-B/unetdl4j/issues/2#issuecomment-663820424).

Are you still working on this part of the issue?

Medha-B commented:
I deleted the dataset...it is no longer present in this branch. To use the code with System.getProperty("user.dir"), one would actually need the dataset to be present in the directory. At this moment, since there is no dataset folder, this code will not actually work. So, I wanted to know if adding the dataset folder again would be alright, without making any changes to the .gitignore file. 
Once I add the dataset folder again, the code will be ready. I will commit it to this branch.
I understand I am being very dense but cannot help it.
Thanks.

kaitoii11 commented:
Gotcha! No worries! 
The dataset should not be maintained in GitHub but the dataset needs to reside on the `dataset` directory in the user's local directory. So what a user should do in order to run your code is as follow.
1. Clone this project
2. Download the dataset to `xxx\unetdl4j\dataset` (You should specify where the user should put the dataset in README.md)
3. Run the code

> I deleted the dataset...it is no longer present in this branch. To use the code with System.getProperty("user.dir"), one would actually need the dataset to be present in the directory. At this moment, since there is no dataset folder, this code will not actually work. So, I wanted to know if adding the dataset folder again would be alright, without making any changes to the .gitignore file.

Yes, you can add the dataset folder. Just keep in mind that dataset folder will not be maintained on git, the user is responsible to maintain it.

Medha-B committed:
Update .gitignore

Medha-B committed:
Create delete.java

Medha-B committed:
Add files via upload

Medha-B committed:
Add files via upload

Medha-B committed:
Add files via upload

Medha-B committed:
Delete delete.java

Medha-B committed:
Modified path again, corrected crossVal for ScoreIterationListener

Medha-B commented:
Kindly check and let me know in case of any dissatisfaction.

kaitoii11 commented on src/main/java/org/sbml/spatial/segmentation/TrainUnetModel.java:
```diff
@@ -111,14 +114,14 @@ public static void main(String[] args) {
+      //ComputationGraph model  = UNet.builder().updater(new Adam(1e-4)).build().init();
+      //ComputationGraph model  = UNet.builder().updater(new Adam(new MapSchedule(ScheduleType.ITERATION, learningScheduleMap))).build().init();
       
       
       //To change the number of input channels from 3 to 1
-        //ZooModel unet = UNet.builder().build();
-	    //unet.setInputShape(new int[][]{{1, 128, 128}});
-	    //ComputationGraph model = (ComputationGraph) unet.init();
+        ZooModel unet = UNet.builder().build();
+	    unet.setInputShape(new int[][]{{1, 128, 128}});
+	    ComputationGraph model = (ComputationGraph) unet.init();
```
Why did we change this part of the code?
Is it any relevant to the issue?

Medha-B replied:
There was an error...I had added this bit earlier for setting the number of input channels to 1. 
Not relevant to path....just to avoid an error that comes with using 1 input channel.

kaitoii11 replied:
Did this error occur due to the changes you made on this branch that are relevant to the issue? If so, it makes sense to incorporate this change. If not, this should be done in a separate branch.

Medha-B replied:
So, should I undo this change and make a separate pull request? Because it is not relevant to this particular issue (regarding path).

kaitoii11 replied:
Yes.

Medha-B replied:
Ok...will do so.

kaitoii11 commented on src/main/java/org/sbml/spatial/segmentation/LearningKfold.java:
```diff
@@ -61,19 +61,21 @@ public static void main(String[] args) {
                 pathToImage = args[0];
             } else {
-                pathToImage = home + File.separator + "Raw images" + File.separator + "F01_621w1_crop13.tif";
+                pathToImage = directory + File.separator + "dataset" + File.separator + "Raw images" + File.separator + "F01_621w1_crop13.tif";
```
Maybe better to assign a new variable that stores the path to dataset since it comes up often.

Medha-B replied:
Sure, will do so.

kaitoii11 commented on src/main/java/org/sbml/spatial/segmentation/crossVal.java:
```diff
@@ -34,13 +34,14 @@ public static void main(String[] args) {
             UnetPathLabelGenerator labeler = new UnetPathLabelGenerator();
-            //File rootDir = new File("C:\\Users\\Subroto\\small_dataset");
-            File rootDir = new File(home + File.separator + "small_dataset");
+            //File rootDir = new File("C:\\Users\\Subroto\\unetdl4j\\dataset\\small_dataset");
```
Probably remove this line since we won't be needing it

Medha-B replied:
Just kept it as an indicator for myself. Will remove it.

kaitoii11 commented:
I'm confused. I thought we were ignoring all the dataset images which are written in gitignore. But the images are added in the repo.

Medha-B commented:
I might have done something wrong with the .gitignore file. I really do not know the format.

kaitoii11 commented:
Ok. Let's discuss this in tonight's meeting.

kaitoii11 commented:
close #7

Medha-B commented:
I have been trying to work the .gitkeep thing out, but even after following all the steps, the images are still visible.

kaitoii11 commented:
Could you write out all the steps you've done line by line?
I'll retrace your steps and try to figure out what might be the problem

Medha-B commented:
First I created a folder dataset in the directory of my dummy repository. Then I added the small_dataset folder to this dataset. Within the small_dataset folder, I created a .gitkeep file using the command prompt.
Then on the Git Bash, I wrote git add dataset/small_dataset, next git commit, and then push.
After this, I edited the .gitignore file and added the lines 
# Dataset
*.tiff
*.tif
and saved it. Next on the git bash, command git add .gitignore, git commit, and push.
The dataset is visible and the images are also visible.
https://github.com/Medha-B/learnBash

Medha-B commented:
First I created a folder dataset in the directory of my dummy repository. Then I added the small_dataset folder to this dataset. Within the small_dataset folder, I created a .gitkeep file using the command prompt.
Then on the Git Bash, I wrote git add dataset/small_dataset, next git commit, and then push.
After this, I edited the .gitignore file and added the lines
"# Dataset
*.tiff
*.tif"
(without the quotes) and saved it. Next on the git bash, command git add .gitignore, git commit, and push.
The dataset is visible and the images are also visible.
https://github.com/Medha-B/learnBash

kaitoii11 commented:
Where in the steps did you add/commit the images?

Medha-B commented:
When I first created the dataset folder and added the small_dataset, I actually added all the images present in the small_dataset. then I added the .gitkeep file. Next, I wrote git add dataset/small_dataset on the bash.

kaitoii11 commented:
So, does that mean you commited the dataset folder & images before editing .gitignore?

Medha-B commented:
Yes. 
I followed these steps given by Akira Sir on slack
"create folder dataset/small_dataset.
create folder dataset/Raw images
create folder dataset/Inference(300x100)
create .gitkeep (an empty file) in the above three directories.
git add dataset/small_dataset
git add dataset/Raw images/
git add dataset/Inference(300x100)
git commit dataset and push
And then edit .gitignore add following lines:
# Dataset
*.tiff
*.tif
then
git add .gitignore
git commit and push!"
Only instead of making 3 directories, I made 1 directory.

kaitoii11 commented:
OK.  Then can you try again from the beginning? This time, don't add the images until you edit the .gitignore. Just create the directory itself and .gitkeep.
Also,  run `git rm -rf dataset` before your first step so you can have a clean start.

Medha-B commented:
Sure. Will report back on the results soon.

Medha-B committed:
fresh start

Medha-B committed:
Adding the empty dataset folder

Medha-B committed:
Modified gitignore

Medha-B committed:
Restored to initial config: 117-125 (TrainUnetModel)

Medha-B committed:
Assigned variable to store path to dataset

Medha-B commented:
It looks like adding images after editing .gitignore worked. Please check and let me know in case of any discrepancies.
Also, I figured that in UnetTrainAndSave and in UnetLoadAndTest, we have to save the model weights and then retrieve them respectively. So, should we include a folder similar to dataset that keeps the model weights without tracking them on git?
Similarly, in the case of model inference, we get an image as output. Could we save it in the somewhere in the dataset folder or maybe a separate folder?

Medha-B commented:
Again, many thanks for helping.

kaitoii11 commented:
> I figured that in UnetTrainAndSave and in UnetLoadAndTest, we have to save the model weights and then retrieve them respectively. So, should we include a folder similar to dataset that keeps the model weights without tracking them on git?

OK. Let's create a different issue for this one! And yes we should create a folder that is not going to be tracked.


> Similarly, in the case of model inference, we get an image as output. Could we save it in the somewhere in the dataset folder or maybe a separate folder?

Yes. A separate folder would be ideal.

Medha-B commented:
Should I merge this pull request now? Or is it alright if I make new branches with this branch (modify_Path) as the origin for other issues?

kaitoii11 commented:
Yes. But before you do so, could you format your code?
https://stackoverflow.com/questions/15655126/how-to-auto-format-code-in-eclipse

>  Or is it alright if I make new branches with this branch (modify_Path) as the origin for other issues?

You should probably create from masters.

Medha-B committed:
Formatted
